addons/web_editor/static/tests/html_field_tests.js

```diff
@@ -797,4 +797,62 @@ QUnit.module("WebEditor.HtmlField", ({ beforeEach }) => {
         assert.strictEqual(p.innerText.replaceAll('\u200B', ''), 'New label',
             "The link's label should be updated");
     });
+
+    QUnit.module("Dropdown");
+
+    QUnit.test('Close dropdown on colorpicker hide', async function (assert) {
+        assert.expect(4);
+        serverData.models.partner.records = [{ id: 1, txt: "<p>first</p>" }];
+        const mockRPC = async function (route, args) {};
+        // Add the ajax service (legacy), because wysiwyg RPCs use it.
+        patchWithCleanup(legacyEnv, {
+            services: {
+                ...legacyEnv.services,
+                ajax: {
+                    rpc: mockRPC,
+                },
+            }
+        });
+        let wysiwyg;
+        const wysiwygPromise = makeDeferred();
+        patchWithCleanup(HtmlField.prototype, {
+            async startWysiwyg() {
+                await this._super(...arguments);
+                wysiwyg = this.wysiwyg;
+                wysiwygPromise.resolve();
+            },
+        });
+
+        await makeView({
+            type: 'form',
+            resId: 1,
+            resModel: 'partner',
+            serverData,
+            arch: '<form>' +
+                    '<field name="txt" widget="html" />' +
+                '</form>',
+        });
+        await wysiwygPromise;
+        const editor = wysiwyg.odooEditor;
+        const editable = editor.editable;
+        const paragrah = editable.querySelector("p");
+        const pText = paragrah.childNodes[0];
+        const toolbar = document.querySelector("#toolbar");
+        setSelection(pText, 1, pText, 3);
+
+        await new Promise(resolve => setTimeout(resolve, 50));
+        assert.ok(toolbar.style.visibility === "visible");
+        toolbar.classList.remove('d-none');
```

Comment:
Why are you testing that the toolbar is visible, then manually removing a class that - if there - makes the toolbar invisible (therefore invalidating the test just before)?

Reply:
The problem here is that `d-none` and `visibility` are handled by different things in the code, The one I'm interested in is `visibility` but it stays with `display=none` so I cant' click on the button on it, so I tried to make a hacky test , idk if it's the best idea

Reply:
The problem with this is that it looks to me like you're testing if the toolbar is visible and the test passes while the toolbar actually isn't. So either you say that's normal and then it should be explicit (adding a test that `d-none` is there for instance), or it's not normal and that should be fixed or at least there should be a comment with a todo.

Reply:
I just realized that I can skip the visibility check on the click on the button which solves my problem, no need to remove `d-none` anymore